### ISSUE TITLE
connection issue when controlling multiple plotters with the same computer

### ISSUE THREAD

**Issue by LiciaHe** [evil-mad/axidraw#81]
Hi, 
I have been trying to connect multiple plotters (2 to be exact) to one mac and I ran into connection issues in both Inkscape and the python API. I tried both the port name (e.g., "/dev/....") and the nickname mode. Nevertheless, none of these worked. When using "list_names", I can see both of my plotters, but whenever I call "connect", it always failed. 

I traced back with some printing statements and the issue I found is in the **ebb_serial.py**  file. 

https://github.com/evil-mad/axidraw/blob/b14acfa20ebe61d283f28f2e347202d4bea58de5/cli/pyaxidraw/ebb_serial.py#L91

Line 91 and Line 92:

Instead of 
```
needle = 'SER=' + port_name     # pyserial 3
needle2 = 'SNR=' + port_name    # pyserial 2.7
```
It should be 

```
needle = 'SER==' + port_name     # pyserial 3
needle2 = 'SNR==' + port_name    # pyserial 2.7
```

I didn't test this on Windows or Linux, but adding the "=" solved my issue. 

Thanks for the awesome API btw.

**Comment by oskay** [comment]
This does not sound right to me.

I'm running some tests on a Mac, using two EBBs with names "Chicago" and "Detroit".

Under either Inkscape 0.92/Python 2 or Inkscape 1.0/Python 3, I can print out what the three port name items are, and get the same results:

```
p0: /dev/cu.usbmodemchicago1
p1: eibotboard_chicago
p2: usb vid:pid=04d8:fd92 ser=chicago location=20-6
p0: /dev/cu.usbmodemdetroit1
p1: eibotboard_detroit
p2: usb vid:pid=04d8:fd92 ser=detroit location=20-4
```

Testing with EBBs named "/dev/cu.usbmodem14601" and "/dev/cu.usbmodem14401", I get the following:

```
p0: /dev/cu.usbmodem14601
p1: eibotboard_
p2: usb vid:pid=04d8:fd92 location=20-6
p0: /dev/cu.usbmodem14401
p1: eibotboard_
p2: usb vid:pid=04d8:fd92 location=20-4
```

If a name "detroit" is entered for the EBB to use, then we search for the several possible strings, including
* needle: `ser=detroit`
* needle2: `snr=detroit`

The search code begins as follows:

```
                if needle in p2:
                    return port[0]  # Success; EBB found by name match.
                if needle2 in p2:
                    return port[0]  # Success; EBB found by name match.
                if needle3 in p1:
                    return port[0]  # Success; EBB found by port match.
```

Breaking this down, the searches would go as follows:


`                if needle in p2:`

Our value `needle` is `ser=detroit`.  If p2 is `usb vid:pid=04d8:fd92 ser=detroit location=20-4`, then we will successfully match the port name and locate the desired port.

If you change it so that p2 is `usb vid:pid=04d8:fd92 ser==detroit location=20-4`, there will not be a match. 


It may well be the case that you have an issue preventing you from identifying the right serial port, but changing those `=` signs to `==` does not make sense as a solution.

**Comment by LiciaHe** [comment]
Interesting. I don't know what's causing the difference between your machine and mine, but here's what I see:
```
About to test serial port: doggy
doggy ebb
needle ser==doggy
needle2 snr=doggy
needle3 (doggy)
p0 /dev/cu.bluetooth-incoming-port
p1 n/a
p2 n/a
p0 /dev/cu.bluetooth-modem
p1 n/a
p2 n/a
p0 /dev/cu.usbmodem=kitt1
p1 eibotboard,=kitty
p2 usb vid:pid=04d8:fd92 ser==kitty location=20-2
p0 /dev/cu.usbmodem=dogg1
p1 eibotboard,=doggy
p2 usb vid:pid=04d8:fd92 ser==doggy location=20-1
```

Because I see 
`p2 usb vid:pid=04d8:fd92 ser==kitty location=20-2`
 and 
`p2 usb vid:pid=04d8:fd92 ser==doggy location=20-1`

I changed needle and needle2 into == and it did solve the problem.

**Comment by oskay** [comment]
Wow-- *that* is interesting!  It's acting as though your EBBs are named `=kitty` and `=doggy`, rather than if they were named `kitty` and `doggy`

When you run **Extensions > AxiDraw Utilities > AxiDraw Naming... > Read AxiDraw Name**, can you please tell me what it returns?

Please also let me know:
* Inkscape version
* Python version
* pyserial version
* macOS version

**Comment by LiciaHe** [comment]
Ha! You are right! Thanks for pointing that out. I could have noticed that and just tried "=kitty". 
Now that I am thinking about it, I probably set it wrong. I went back to[ the documentation](https://axidraw.com/doc/py_api/#read_name) 
```
ad.options.mode = "manual"
ad.options.manual_cmd = "write_nameAxiDraw 17"
```
and realize that I probably wrote ` "write_nameAxiDraw 17"` as `"write_name=AxiDraw 17"`. No wonder why it includes an equal sign.  Although I did use `list_names` multiple times, I just assumed that equal sign is part of the printing but not the name. 

And yes, I ran `list_names` again, the return is indeed =kitty and =doggy. 

I'd say this is definitely not a bug of the API but me skipping the doc example :) Thanks for your help. This would totally bug me for a long time without your help.
